model/operation.cc

```diff
@@ -360,13 +386,24 @@ namespace minsky
       {
         drawPorts(cairo);
         displayTooltip(cairo,tooltip);
+        if (onResizeHandles) drawResizeHandles(cairo);
       }
 
     cairo_new_path(cairo);
     clipPath.appendToCurrent(cairo);
     cairo_clip(cairo);
     if (selected) drawSelected(cairo);
   }
+  
+  void OperationBase::resize(const LassoBox& b)
+  {
+    float w=iWidth(), h=iHeight(), invZ=1/zoomFactor();
+    iWidth(std::abs(b.x1-b.x0)*invZ);
+    iHeight(std::abs(b.y1-b.y0)*invZ);
+    moveTo(0.5*(b.x0+b.x1), 0.5*(b.y0+b.y1));
+    bb.update(*this);	  
```

Comment:
Given that iWidth and iHeight call bb.update, this is a redundant third call. May need to perform moveTo before the calls to iWidth and iHeight.